### PR TITLE
Request Wrapper: Only Create Elements If Needed

### PR DESCRIPTION
**Pull request by SyntaxNode**
This PR change the behavior of the RequestWrapper to only create new objects if there a non-empty extension object. I also fixed some misc marhsal issues that I discovered through the rebuild ext methods.

### CODE REVIEW

File: openrtb_ext/request_wrapper.go [hunk 1]
```diff
@@ -780,7 +832,7 @@ func (re *RegExt) SetUSPrivacy(uSPrivacy string) {
 type SiteExt struct {
 	ext      map[string]json.RawMessage
 	extDirty bool
-	amp      int8
+	amp      *int8
```
Comment (AlexBVolcy) [inline openrtb_ext/request_wrapper.go:835]: We spoke about this yesterday, but I just wanted to confirm: this was changed to a pointer so we could check if this `amp` variable is empty?

**Review by AlexBVolcy — COMMENTED** [review]
Looks good! Just left some minor comments

File: openrtb_ext/request_wrapper.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ type RequestWrapper struct {
 	sourceExt  *SourceExt
 }
 
+const jsonEmptyObjectLength = 2
```
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply openrtb_ext/request_wrapper.go:44]: Yes, exactly. It's the only reason a json object will be 2 characters.

File: openrtb_ext/request_wrapper.go [hunk 3]
```diff
@@ -780,7 +832,7 @@ func (re *RegExt) SetUSPrivacy(uSPrivacy string) {
 type SiteExt struct {
```
Comment (mansinahar) [inline openrtb_ext/request_wrapper.go:832]: Not really related to your changes but wondering if there's any difference between `SiteExt` that's defined here vs `ExtSite` defined in `openrtb_ext/site.go`? The latter doesn't really seem to be used anywhere.. perhaps we can remove it as part of a separate PR?

**Review by mansinahar — APPROVED** [review]

**Review by AlexBVolcy — APPROVED** [review]

**Pull request merged**